arch/arm/include/aarch32/cortex_m/tz.h

```diff
@@ -306,6 +307,46 @@ typedef void __attribute__((cmse_nonsecure_call)) (*tz_ns_func_ptr_t) (void);
 
 #endif /* CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS */
 
+/**
+ * @brief Name of a non-secure entry function if it is to be wrapped.
+ */
+#define __TZ_NSC_NAME(name) nsc_ ## name
+
+/**
+ * @brief Create a thread safe wrapper function for an NS entry function.
+ *
+ * This locks the scheduler before calling the function by wrapping the NS entry
+ * function in @ref k_sched_lock / @ref k_sched_unlock, using
+ * @ref Z_ARM_WRAP_FUNC.
+ *
+ * In secure code:
```

Comment:
This section does not belong here, and is really confusing. This "wrapping" is only meant to be called in non-secure domain, right? I suggest that we add this section as a prerequisite: 
For example

The non-secure entry function needs to be defined (in the Secure firmware) using the __TZ_NSC_NAME macro.